packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx

```diff
@@ -1,70 +1,118 @@
-import { Component, h, Prop, Event, type EventEmitter, State } from '@stencil/core';
+import { Component, h, Prop, Event, type EventEmitter } from '@stencil/core';
 import type { IconDescription } from '@eventstore-ui/components';
 import { ICON_NAMESPACE } from '../../icons/namespace';
-import type { FieldChange } from '../../types';
+import type { FieldChange, ValidationMessages } from '../../types';
+import { Field } from '../Field/Field';
 
-/** A multi-checkbox component */
+/**
+ * A multi-checkbox component
+ * @part checkbox-field - Checkbox Field.
+ */
 @Component({
     tag: 'es-multi-checkbox',
     styleUrl: 'es-multi-checkbox.css',
     shadow: true,
 })
 export class EsMultiCheckbox {
     /** Emitted when the value of the field is changed. */
-    @Event({ bubbles: true }) fieldchange!: EventEmitter<FieldChange<Record<string, boolean>>>;
+    @Event({ bubbles: true }) fieldchange!: EventEmitter<
+        FieldChange<string | null>
```

Comment:
Why did this get changed to `string / null`? I don't see anything about it in the commit message. I think the previous interface seemed nicer to use, did something not work with it?

Reply:
Ah, yeah, that's for nested forms, maybe there's a better way to do it, but best not to mess with it now.

My only concern with that interface is that it will break if we pass an option with a new-line in it. It's probably an unfounded concern, but might bite us later.

How about `Set<string>`? if it's in the set, it's checked, if not, it's not. 

The change function becomes a bit simpler also:

```
        const { value, checked } = e.target as HTMLInputElement;

        if (!value) return;

        let newValue = new Set(this.value);

        if (checked) {
            newValue.add(value);
        } else {
            newValue.delete(value);
        }

        this.fieldchange.emit({
            name: this.name,
            value: newValue,
        });
 ```
 
 a `Map<string, boolean>` could also work:
 
 ```
         const { value, checked } = e.target as HTMLInputElement;

        if (!value) return;

        let newValue = new Map(this.value);

        
        newValue.set(value, checked); 


        this.fieldchange.emit({
            name: this.name,
            value: newValue,
        });
 ````
 
 But seeing if it's checked on the outside is a bit more complicated. 
 
 ```
 const checked = data.thingAsMap.get("name") ?? false;
 ```
 
 vs
 
  ```
 const checked = data.thingAsSet.has("name");
 ```

Reply:
I believe Set and Map are also object, and will have the same issue

Reply:
eugh, you're right. This should be all we need to allow it to work:

https://github.com/EventStore/Design-System/pull/386

Reply:
I've also updated this because it appears to have the same logic.
![image](https://github.com/user-attachments/assets/f40a9d89-9c6a-4cc5-a690-c137940e6314)